core/block.js

```diff
@@ -1106,7 +1107,7 @@ Blockly.Block.prototype.renameVarById = function(oldId, newId) {
 Blockly.Block.prototype.getFieldValue = function(name) {
   var field = this.getField(name);
   if (field) {
-    return field.getValue();
+    return String(field.getValue());
```

Comment:
This is an interesting one. field.getValue() can be of any type internally, but getFieldValue promises a string. Casting to string might not be desired, we could instead return a ``{*}``. Thoughts @rachel-fenichel?

Reply:
I think it needs to return `{*}` because `String()` will break object values.

Reply:
Yep, that’s a fair point. I’ll change it to return {string} instead